### DISCUSSION TITLE
Simulation of an electrode

### DISCUSSION THREAD

**Discussion by Leoearth** [Electrostatics/apbs#295] [Q&A]
How can i simulate the presence of an electrode in APBS?

**Comment by sobolevnrm** [comment]
It might be possible to emulate this with [bcfl map](https://apbs.readthedocs.io/en/latest/using/input/old/elec/bcfl.html), although I haven't tried it or thought through the implications entirely.

**Comment by cdcooper84** [comment]
Thanks so much for you answer @sobolevnrm! 

We were thinking about using `bcfl map`, and it seems to be reasonable (and similar to what it is done in [APBSmem](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0012722#)). However, we plan now to do nonlinear simulations, then, we're wondering if this approach is still valid. There is a nice [B. Roux paper](https://pmc.ncbi.nlm.nih.gov/articles/PMC1181204/) explaining the external potential, but only covers the linear case, where decoupling terms is pretty straight forward. On the other hand, there is an [APBSmem paper that uses the nonlinear PB](https://www.cell.com/structure/fulltext/S0969-2126(15)00195-1), so we're guessing it should be valid. Do you see any contradiction of using the nonlinear PB along with `bcfl map`?

**Comment by sobolevnrm** [comment]
Hi @cdcooper84 --

Yes, `bcfl map` is the way I would approach this problem.  I'm not sure if APBSmem is actively developed/supported, so I didn't want to recommend it.

Thanks!

**Reply by Leoearth** [reply]
Thank you for your answer.

I have done some calculations of a protein between two electrodes set 140.4 A apart along the x axis. Therefore, i have two surfaces, one at x=0 and the other at 140.4 A. I simulated them by a dx file and then a bcfl map boundary condition to have a constant potential. 

I computed the electrostatic component of the electrode-protein interaction by taking the difference between the solvation energy of the protein close to one electrode and the solvation energy of the protein in the middle of the domain, far away from both electrodes. 
I'm puzzled because for most cases (eg different pH values and different protein orientations on the electrode and different electrode potentials and different proteins) I obtained a repulsive interaction between the protein and the electrode. For example, there is a repulsive interaction even when a negatively charge protein interacts with an electrode at a positive potential.

I attach a zip directory with the sample calculation (I had to delete the dx file from the directory because it was too large to upload, but I can provide it if required). Can you please have a look and see if you can spot anything which is obviously wrong?

Thank you,

Leo

[Gox_pH7_pot_10.zip](https://github.com/user-attachments/files/18736325/Gox_pH7_pot_10.zip)
